exercise3.js

```diff
@@ -85,32 +127,116 @@
 // 7.a - Write a function to find the maximum numerical value of the given array.  Get rid of any non numerical values.  Convert the strings that are numbers to an actual number data type.  ("one" => 1) ("1" => 1).  Use array methods to perform this task.  
 const numbers = [2,23,1,2,1,1,1,2,2.5,20,200,2000,,{k:1},20000,19999,1878,140,23,4,"sk",true,true,"true-dat","nice","one","two","three","3","tea",[]];
 
-function maxNumber(numbers) {
-    //your code...
-}
+let max = (arr) => {
+  let nums = [];
+   arr.map(item=>{
+       //checks specific values than checks if typeof is number if so pushes to array
+    if (item == 'one') {
+      nums.push(1);
+    }
+    if (item == 'two') {
+      nums.push(2);
+    }
+    if(item=='3'){
+        nums.push(3)
+    }
+    if (item == 'three') {
+      nums.push(3);
+    }
+    if (typeof item === 'number') {
+      nums.push(item);
+    }
+    })
+  return Math.max(...nums);
+};
 
+max(numbers)
 // 7.b -Write a function that sorts the given numbers array.  Allow the function to sort the array in descending order
 
-function sortNums(numbers,desc=false) {
-    //your code...
+
+let sorted = (arr, mer) => {
+  let nums = [];
+  arr.map((item) => {
+      //same as above
+    if (item == 'one') {
+      nums.push(1);
+    }
+    if (item == 'two') {
+      nums.push(2);
+    }
+    if (item == '3') {
+      nums.push(3);
+    }
+    if (item == 'three') {
+      nums.push(3);
+    }
+    if (typeof item === 'number') {
+      nums.push(item);
+    }
+  });
+  console.log(mer(nums));
 };
 
+//my implementation of merge sort but it does it backwards
+let reverseMergeSort = (arr) => {
+    //checks if array length is 1 if so returns it
+  if (arr.length === 1) {
+    return arr;
+  }
+  //gets middle index
+  let middle = Math.floor(arr.length / 2);
+  //cuts left side
+  let left = arr.slice(0, middle);
+  //cuts right side
+  let right = arr.slice(middle);
+  //recursevly  does above function
+  let sortedLeft = reverseMergeSort(left);
+  let sortedRight = reverseMergeSort(right);
+  return merge(sortedLeft, sortedRight);
+};
+//helper function to main function
+let merge = (left, right) => {
+
+  let results = [];
+  let leftIndex = 0;
+  let rightIndex = 0;
+  //while the left Index is less than left side so until its fully splitted 
+  while (leftIndex < left.length && rightIndex < right.length) {
+      //if number in the left is less than number in the right push right THIS IS WHAT PRETTY MUCH MAKES IT GO FROM BIG TO LOW
+    if (left[leftIndex] < right[rightIndex]) {
+      results.push(right[rightIndex]);
+      rightIndex++;
+    } else {
+        //SECOND PART OF PUSHING 
+      results.push(left[leftIndex]);
+      leftIndex++;
+    }
+  }
+  //start adding all the pieces that were splittled above
+  return results.concat(left.slice(leftIndex)).concat(right.slice(rightIndex));
+};
 
+sorted(numbers, reverseMergeSort);
 
```

Comment:
awesome job looking into the merge sort algoirthm.  The problem was asking to provide two options to sort, depending on if the descending boolean was true or false.  so your function should be able to sort in ascending or descending orders, depending on the 2nd parameter.